### DISCUSSION TITLE
Trying to see if it's possible to decorate typescript's autocomplete using a custom vscode extension

### DISCUSSION THREAD

**Discussion by YitzchakSviridyukTW** [microsoft/vscode-discussions#663] [Extension Development QnA] [answered]
I have a const that I'm using as a type for one of the properties in my interfaces:
type ColorArray = Tuple<string, 10>;

```ts
export default [
  rose: [
    '#FFF1F2',
    '#FFE4E6',
    '#FECDD3',
    '#FDA4AF',
    '#FB7185',
    '#F43F5E',
    '#E11D48',
    '#BE123C',
    '#9F1239',
    '#881337',
  ] as ColorArray,
  indigo: [
    '#312E81',
    '#3730A3',
    '#4338CA',
    '#4F46E5',
    '#6366F1',
    '#818CF8',
    '#A5B4FC',
    '#C7D2FE',
    '#E0E7FF',
    '#EEF2FF',
  ] as ColorArray,
  // etc.
} as const;
```

I use it for this type:
```ts
export type FormattedColor = `${keyof typeof colors}.${NumRange<0, 10>}`;
```

This type gives me 150 options - 10 for each of 15 different colors.  Something like "blue.6", "orange.2", etc., where blue is the color and 6 is the shade.
These are all valid colors for my interfaces.  Typescript gives me autocomplete for these already.  However, I want the color that each of these represents to be next to the automatic suggestions of typescript, like in the screenshot.

I made a simple custom vscode extension for this, but it doesn't work with typescript, meaning I'm seeing typescript's suggestions first, then I see the extensions suggestions with the color afterwards.  Is there a way to make some sort of extension that can hook into the actual typescript suggestions?  If so, how?

Would be grateful for any input.  Thanks :) 

![image](https://github.com/microsoft/vscode-discussions/assets/121752336/cd157a3d-cde1-4e5b-8195-3e231e428ab1)

**Comment by r3m0t** [comment, marked as answer]
https://github.com/microsoft/TypeScript/wiki/Writing-a-Language-Service-Plugin

This could either provide the correct response or suppress the response so that the result from your extension is the only one shown.

**Reply by YitzchakSviridyukTW** [reply]
Oh!  This looks good.  I'm going to play with it a bit then get back to you.  Thanks!
